### PR TITLE
📝 Split Button story into sub-directories

### LINKED ISSUE #2509: Split Button .Group & .Toggle into sub-directories
Suggesting to split `Button` docs into  sub-directories. There’s a bit too much in the docs after adding `ButtonGrout` & `ToggleButton`.

### PR DESCRIPTION
**Pull request by martalalik**
Resolves #2509

### CODE REVIEW

File: packages/eds-core-react/src/components/Button/ButtonGroup/ButtonGroup.docs.mdx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## Examples
+
+### Horizontal
+The `Button.Group` is displayed horizontally by default.
+
+<Story id="inputs-buttons-group--horizontal" />
+
+### Vertical
+The `Button.Group` can be displayed vertically using the `vertical` prop.
```
Comment (mimarz) [inline packages/eds-core-react/src/components/Button/ButtonGroup/ButtonGroup.docs.mdx:47]: Suggest merging these into one story and just calling it "orientation"? Then just have both the horizontal and vertical in the same story.

**Review by mimarz — CHANGES_REQUESTED** [review]
1. Suggest just sticking to one variant for the "Button.Group" examples. Having one unlike the other makes it look like the toggle button when its not... Suggest just sticking to default `<Button>`

2. Maybe just call the parent directory just "Button" instead of "Buttons"? "Button" felt more natural and in line with the sub-components and such

**Comment by martalalik** [comment]
> 2. Maybe just call the parent directory just "Button" instead of "Buttons"? "Button" felt more natural and in line with the sub-components and such

Definitely 😅

**Comment by oddvernes** [comment]
I noticed this warning when making a build (from develop not this branch):
```
(!) Circular dependency
src/components/Button/index.ts -> src/components/Button/ToggleButton/ToggleButton.tsx -> src/components/Button/index.ts
```

File: packages/eds-core-react/src/components/Button/ButtonGroup/ButtonGroup.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  ],
+} as ComponentMeta<typeof Button.Group>
+
+export const Introduction: Story<ButtonGroupProps> = (args) => {
+  return (
+    <Button.Group {...args} aria-label="primary actions">
+      <Button>Contained</Button>
+      <Button variant="outlined">Outlined</Button>
+      <Button variant="outlined">Outlined</Button>
+      <Button variant="outlined">Outlined</Button>
```
Comment (mimarz) [inline packages/eds-core-react/src/components/Button/ButtonGroup/ButtonGroup.stories.tsx:41]:
```suggestion
      <Button>Button</Button>
      <Button>Button</Button>
      <Button>Button</Button>
      <Button>Button</Button>
```

File: packages/eds-core-react/src/components/Button/ButtonGroup/ButtonGroup.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    </Button.Group>
+  )
+}
+
+export const Horizontal: Story<ButtonGroupProps> = () => (
+  <Button.Group aria-label="primary actions">
+    <Button>Contained</Button>
+    <Button variant="outlined">Outlined</Button>
+    <Button variant="outlined">Outlined</Button>
+    <Button variant="outlined">Outlined</Button>
```
Comment (mimarz) [inline packages/eds-core-react/src/components/Button/ButtonGroup/ButtonGroup.stories.tsx:51]:
```suggestion
    <Button>Button</Button>
    <Button>Button</Button>
    <Button>Button</Button>
    <Button>Button</Button>
```

File: packages/eds-core-react/src/components/Button/ButtonGroup/ButtonGroup.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  </Button.Group>
+)
+
+export const Vertical: Story<ButtonGroupProps> = () => (
+  <Button.Group aria-label="vertical actions" vertical>
+    <Button>Contained</Button>
+    <Button variant="outlined">Outlined</Button>
+    <Button variant="outlined">Outlined</Button>
+    <Button variant="outlined">Outlined</Button>
```
Comment (mimarz) [inline packages/eds-core-react/src/components/Button/ButtonGroup/ButtonGroup.stories.tsx:60]:
```suggestion
    <Button>Button</Button>
    <Button>Button</Button>
    <Button>Button</Button>
    <Button>Button</Button>
```

File: packages/eds-core-react/src/components/Button/ButtonGroup/ButtonGroup.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    setDensity('compact')
+  }, [density])
+
+  return (
+    <EdsProvider density={density}>
+      <Button.Group aria-label="compact actions">
+        <Button>Contained</Button>
+        <Button variant="outlined">Outlined</Button>
+        <Button variant="outlined">Outlined</Button>
+        <Button variant="outlined">Outlined</Button>
```
Comment (mimarz) [inline packages/eds-core-react/src/components/Button/ButtonGroup/ButtonGroup.stories.tsx:135]:
```suggestion
    <Button>Button</Button>
    <Button>Button</Button>
    <Button>Button</Button>
    <Button>Button</Button>
```

**Review by mimarz — CHANGES_REQUESTED** [review]

**Review by oddvernes — APPROVED** [review]
LGTM, just fix linting errors in ButtonGroup.stories

**Review by mimarz — APPROVED** [review]
LGTM 👍

**Pull request merged**